File: LeetCode/cpp/1005_Maximize_Sum_Of_Array_After_K_Negations.cpp
class Solution {
public:
	int largestSumAfterKNegations(vector<int>& nums, int k) {
		int answer = 0;
		priority_queue<int, vector<int>, greater<>> pq(nums.begin(), nums.end());

		while (k > 0) {
			int n = pq.top();
			pq.pop();

			pq.push(-1 * n);

			k--;
		}

		while (!pq.empty()) {
			answer += pq.top();
			pq.pop();
		}

		return answer;
	}
};

File: LeetCode/java/1005_Maximize_Sum_Of_Array_After_K_Negations.java
import java.util.PriorityQueue;

class Solution {
    public int largestSumAfterKNegations(int[] nums, int k) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        for (int n : nums) {
            pq.add(n);
        }

        while (k > 0) {
            int n = pq.poll();

            pq.add(-1 * n);

            k--;
        }

        while (!pq.isEmpty()) {
            answer += pq.poll();
        }

        return answer;
    }
}